internal/client/service/teams.go

```diff
@@ -163,12 +164,30 @@ func (c *TeamsClient) GetTeamDeploymentGrantByTeamAndDeploymentId(ctx context.Co
 }
 
 func (c *TeamsClient) CreateOrUpdateTeamDeploymentGrant(ctx context.Context, teamId string, deploymentId int, grant schema.PermissionGrant, locationGrants []schema.LocationScopedGrant) (schema.ScopedPermissionGrant, error) {
-	// TODO: check if team exists and check if deployment exists => to return specific errors
-	// TODO: validate sure grant >= location grant and location grant cant be viewer or agent
 	locationGrantsInput := make([]schema.LocationScopedGrantInput, 0, len(locationGrants))
 
+	deploymentGrantIdx := utils.IndexOf(types.DeploymentGrantEnumValues(), string(grant))
+
 	for _, locationGrant := range locationGrants {
-		// TODO check if code location exists
+		locationGrantIdx := utils.IndexOf(types.DeploymentGrantEnumValues(), string(locationGrant.Grant))
+
+		if deploymentGrantIdx >= locationGrantIdx {
+			return schema.ScopedPermissionGrant{}, &types.ErrInvalid{
+				What:    "TeamDeploymentGrant",
+				Message: "LocationGrant can't be higher than DeploymentGrant",
```

Comment:
```suggestion
				Message: "LocationGrant can't be less permissive than DeploymentGrant",
```

Maybe not bad to put logic to compare grant values in a helper fn